transport/server.hh

```diff
@@ -177,6 +178,8 @@ private:
         future<> process();
         future<> process_request();
         future<> shutdown();
+        void fill_client_data(client_data& cd) const;  // Arg here must go by ref (because fwd decl)
```

Comment:
It's nicer to return a value instead of using an out-parameter.

Reply:
Done. (although that introduces one more `#include` which I tried to avoid)

Reply:
Why do you need a #include? returning a forward declared class should work, no?